### PR TITLE
otelcol: add systemd logs pipeline

### PR DESCRIPTION
**Pull request by pmalek-sumo**
This requires new release of https://github.com/SumoLogic/sumologic-otel-collector/ which will include the already merged changes and additionally the following unmerged PRs:

* https://github.com/SumoLogic/sumologic-otel-collector/pull/236
* https://github.com/SumoLogic/sumologic-otel-collector/pull/239

### CODE REVIEW

File: deploy/helm/sumologic/values.yaml [hunk 1]
```diff
@@ -2821,12 +2834,60 @@ otelcol:
...
+                record_attributes:
+                  - key: fluent.tag
+                    value: "containers.var.log.containers.collection-sumologic-otelcol-logs.*"
+          filter/fluent_tag_receiver_mock_container:
+            logs:
+              exclude:
+                match_type: regexp
+                record_attributes:
+                  - key: fluent.tag
+                    value: "containers.var.log.containers.receiver-mock.*"
```
Comment (sumo-drosiek) [inline deploy/helm/sumologic/values.yaml:2869]: This shouldn't go in this PR. You can add it to vagrant's local.conf

File: deploy/helm/sumologic/values.yaml [hunk 2]
```diff
@@ -2821,12 +2834,60 @@ otelcol:
...
+              - _SYSTEMD_UNIT
+              - _HOSTNAME
+
+          filter/fluent_tag_otelcol_container:
+            logs:
+              exclude:
+                match_type: regexp
+                record_attributes:
+                  - key: fluent.tag
+                    value: "containers.var.log.containers.collection-sumologic-otelcol-logs.*"
```
Comment (sumo-drosiek) [inline deploy/helm/sumologic/values.yaml:2862]: not big change, but still

```suggestion
                    value: "containers\.var\.log\.containers\.collection-sumologic-otelcol-logs.*"
```

**Comment by sumo-drosiek** [comment]
could we order processors in pipeline order. It would be more readable

File: deploy/helm/sumologic/values.yaml [hunk 3]
```diff
@@ -2821,12 +2834,46 @@ otelcol:
...
+                record_attributes:
+                  - key: fluent.tag
+                    value: host\..+
+          filter/include_systemd:
+            logs:
+              include:
+                match_type: regexp
+                record_attributes:
+                  - key: _SYSTEMD_UNIT
+                    value: .+
```
Comment (sumo-drosiek) [inline deploy/helm/sumologic/values.yaml:2876]: It should exclude kubelet:

https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/79a0ea746d53f4f5f4bc3463a0a0c4b8ac299401/deploy/helm/sumologic/conf/logs/fluentd/logs.source.systemd.conf#L5-L8

**Comment by pmalek-sumo** [comment]
> could we order processors in pipeline order. It would be more readable

We could but some of them are reused across pipelines e.g. `attributes/remove_fluent_tag`

**Comment by sumo-drosiek** [comment]
We can move it at beginning of section
with such approach we could comment them

```
## common processors

## containers pipeline processors

## systemd pipeline processors
```

**Review by sumo-drosiek — APPROVED** [review]

**Comment by pmalek-sumo** [comment]
> This requires new release of https://github.com/SumoLogic/sumologic-otel-collector/ ... 

The required release was published as https://github.com/SumoLogic/sumologic-otel-collector/releases/tag/v0.0.27-beta.0

**Pull request merged**